### PR TITLE
Issue 4214 bulk export group observations missed when search parameter indexing enabled

### PR DESCRIPTION
**Pull request by jmarchionatto**

### CODE REVIEW

File: hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/bulk/BulkGroupExportWithIndexedSearchParametersTest.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		mySystemDao.transaction(mySrd, inputBundle);
+
+		// set the export options
+		BulkDataExportOptions options = new BulkDataExportOptions();
+		options.setResourceTypes(Set.of("Patient", "Observation", "Group"));
+		options.setGroupId(new IdType("Group", "G1"));
+		options.setExportStyle(BulkDataExportOptions.ExportStyle.GROUP);
+		options.setOutputFormat(Constants.CT_FHIR_NDJSON);
+
+		BulkExportJobResults jobResults = getBulkExportJobResults(options);
```
Comment (fil512) [inline hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/bulk/BulkGroupExportWithIndexedSearchParametersTest.java:79]: suggestion: rename getBulkExportJobResults to submitBulkExportJobAndReturnResults

File: hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/bulk/BulkGroupExportWithIndexedSearchParametersTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+	private BulkExportJobResults getBulkExportJobResults(BulkDataExportOptions theOptions) {
+		Batch2JobStartResponse startResponse = myJobRunner.startNewJob(BulkExportUtils.createBulkExportJobParametersFromExportOptions(theOptions));
+
+		assertNotNull(startResponse);
+
+		// Run a scheduled pass to build the export
+		myBatch2JobHelper.awaitJobCompletion(startResponse.getJobId());
+
+		await().until(() -> myJobRunner.getJobInfo(startResponse.getJobId()).getReport() != null);
```
Comment (fil512) [inline hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/bulk/BulkGroupExportWithIndexedSearchParametersTest.java:91]: thought: (not for this MR) this looks like a bug.  I would expect transition to complete only after the report is available.

File: hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/bulk/BulkGroupExportWithIndexedSearchParametersTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+		// Iterate over the files
```
Comment (fil512) [inline hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/bulk/BulkGroupExportWithIndexedSearchParametersTest.java:93]: I don't see an iteration here

File: hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SearchWithElasticSearchIT.java [hunk 4]
```diff
@@ -2274,6 +2274,37 @@ void specCase8() {
 		}
 	}
 
+	@Disabled("while finding out strategy for this fix")
```
Comment (fil512) [inline hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SearchWithElasticSearchIT.java:2277]: should this still be in this MR?

**Review by fil512 — APPROVED** [review]
nice clean MR.  great work!

File: hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/bulk/BulkGroupExportWithIndexedSearchParametersTest.java (hunk 2 above)
Comment (jmarchionatto) [reply hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/bulk/BulkGroupExportWithIndexedSearchParametersTest.java:91]: I believe you. Just stole the fragment from other test.

File: hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SearchWithElasticSearchIT.java (hunk 4 above)
Comment (jmarchionatto) [reply hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SearchWithElasticSearchIT.java:2277]: Well. It showed up during this fix, and I din't want to loose the valuable test. I think doesn't hurt.
BTW, we are discussing that problem/fix as I wrote with Gary and Michael

File: hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/bulk/BulkGroupExportWithIndexedSearchParametersTest.java (hunk 3 above)
Comment (jmarchionatto) [reply hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/bulk/BulkGroupExportWithIndexedSearchParametersTest.java:93]: Yep. Leftover from stolen code fragment. Removed

**Pull request merged**